### PR TITLE
[flink] Assign splits with fixed batch size in StaticFileStoreSplitEnumerator

### PR DESCRIPTION
**Pull request by wxplovecc**
### Purpose

Assign all splits in one time will produce exceed `akka.framesize` to exceptions
Add `scan.bounded.splits.size` config option to detemine how many splits will be assign per batch

### Tests

StaticFileStoreSplitEnumeratorTest.testSplitBatch

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java [hunk 1]
```diff
@@ -79,10 +82,15 @@ public void handleSplitRequest(int subtask, @Nullable String hostname) {
         // The following batch assignment operation is for two purposes:
         // To distribute splits evenly when batch reading to prevent a few tasks from reading all
         // the data (for example, the current resource can only schedule part of the tasks).
-        // TODO: assignment is already created in constructor, here can just assign per batch
-        List<FileStoreSourceSplit> splits = pendingSplitAssignment.remove(subtask);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:83]: Can we make `pendingSplitAssignment` to `Map<Integer, Queue<FileStoreSourceSplit>>`?
And poll a batch from it.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java [hunk 2]
```diff
@@ -37,6 +37,9 @@
 public class StaticFileStoreSplitEnumerator
         implements SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> {
 
+    /** Default batch splits size to avoid exceed `akka.framesize`. */
+    private static final int DEFAULT_SPLITS_SIZE = 10;
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:41]: Can we introduce an option for this? Can be an option in `FlinkConnectorOptions`?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java (hunk 1 above)
Comment (wxplovecc) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:83]: +1

File: docs/layouts/shortcodes/generated/flink_connector_configuration.html [hunk 3]
```diff
@@ -26,6 +26,12 @@
             <td>String</td>
             <td>The log system used to keep changes of the table.<br /><br />Possible values:<br /><ul><li>"none": No log system, the data is written only to file store, and the streaming read will be directly read from the file store.</li></ul><ul><li>"kafka": Kafka log system, the data is double written to file store and kafka, and the streaming read will be read from kafka.</li></ul></td>
         </tr>
+        <tr>
+            <td><h5>scan.bounded.splits.size</h5></td>
```
Comment (JingsongLi) [inline docs/layouts/shortcodes/generated/flink_connector_configuration.html:30]: `scan.split-enumerator.batch-size`?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSource.java [hunk 4]
```diff
@@ -94,6 +98,6 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
         }
 
         Snapshot snapshot = snapshotId == null ? null : snapshotManager.snapshot(snapshotId);
-        return new StaticFileStoreSplitEnumerator(context, snapshot, splits);
+        return new StaticFileStoreSplitEnumerator(context, snapshot, splits, splitsSize);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSource.java:101]: Here, we can get `splitBatchSize` from table. (via table.options())

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java [hunk 5]
```diff
@@ -79,10 +86,16 @@ public void handleSplitRequest(int subtask, @Nullable String hostname) {
         // The following batch assignment operation is for two purposes:
         // To distribute splits evenly when batch reading to prevent a few tasks from reading all
         // the data (for example, the current resource can only schedule part of the tasks).
-        // TODO: assignment is already created in constructor, here can just assign per batch
-        List<FileStoreSourceSplit> splits = pendingSplitAssignment.remove(subtask);
-        if (splits != null && splits.size() > 0) {
-            context.assignSplits(new SplitsAssignment<>(Collections.singletonMap(subtask, splits)));
+        Queue<FileStoreSourceSplit> allSubTaskSplits = pendingSplitAssignment.get(subtask);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:89]: minor: rename to `taskSplits`

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java [hunk 6]
```diff
@@ -79,10 +86,16 @@ public void handleSplitRequest(int subtask, @Nullable String hostname) {
...
+        List<FileStoreSourceSplit> batchAssignment = new ArrayList<>();
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/StaticFileStoreSplitEnumerator.java:90]: minor: rename to `assignment`?

**Review by JingsongLi — APPROVED** [review]
Thanks @wxplovecc , +1 for the pr

**Pull request merged**